### DISCUSSION TITLE
Building a sankey diagram

### DISCUSSION THREAD

**Discussion by SimonSimCity** [xyflow/xyflow#3024] [General]
### Please describe the feature that you want to propose

Hey 👋 first of all, this package here looks awesome, and I'd like to use it if possible!

In my project we need a very flexible sankey diagram, which the user should be able to smack around with. This package looks very suitable! I've also seen you can write extensions to it, and I'm wondering, if you see anything that would be blocking or make things incompatible - basically asking if this fits the long-term strategy of this project.

As I see it, the extension would only control the docking-positioning of edges on nodes, according to the thickness of the edges on the node on that direction (e.g. all left edges or right edges should be stacked, not overlapping). It should also be possible to add styles (e.g. color) to individual edges.

Since I couldn't find anything relating this in a quick search, I've opened this feature request. If someone already has set in some work, it would be awesome to hear!

**Comment by moklick** [comment]
What would be the advantage to use React Flow over https://github.com/d3/d3-sankey for example?

**Reply by SimonSimCity** [reply]
We have nodes, which the user should be able to align himself on a grid. To start with its often not known how they're connected, the scale of the connection (width of each connection in a Sankey) will follow also a while later after connecting them. After the scaling is known, the user should still be able to align the nodes himself as he sees it fitting for his purpose.

I'm not very familiar with either libs, but to me, this one looks closer to what I need.

**Reply by moklick** [reply]
If the user should be able to drag nodes, React Flow could be useful :) It should be possible to implement your requirements with React Flow.
